app/controllers/languages_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# LanguagesController: for language functionality
+class LanguagesController < ApplicationController
+  before_action :authenticate_user!
+
+  # Returns all languagaes as an array of arrays where each inner is in the format
+  # ['eng', 'English', supported: {sms: true, phone: true, email: true}]
+  # Example Return: [["eng", "English", {}]...,["zho","Chinese", {}],["zul","Zulu", {}]]
+  def language_data
+    render json: Languages.all_languages
+  end
+
+  # Matches code to display name
+  # params[:language_codes] must be a valid array of language codes (no nils or invalids)
+  # Returns array of the display names for those codes
+  # PARAM EXAMPLE: ['oci', 'tgk']
+  # RETURN EXAMPLE: ['Occitan', 'Tajik']
+  def translate_language_codes
+    language_set_difference = params.require(:language_codes).map(&:to_sym) - Languages.all_languages.keys
+    if language_set_difference.any?
+      error_message = "There was an error looking up the following languages: #{language_set_difference.map(&:to_s)}"
+      render(json: { error: error_message }, status: :unprocessable_entity) && return
```

Comment:
Seeing something funky. Haven't dug into it enough to know if the issue is really here or in the front end or in the migration, but here are the steps I took:
1. On branch `1.28.0` import a patient with an unrecognized language (e.g. 'cheese')
2. Switch to this branch and run migration
3. Visit that patient's page

I get a Toast 422 error, and here is the output:

![image](https://user-images.githubusercontent.com/8235974/116121586-c116dd00-a68e-11eb-9324-2e0cd7f32486.png)

I'll note that if I inspect the patient in the rails console, the language is stored in the database as `""`.

I'll also note that this issue _doesn't_ happen for a patient that was imported with _no_ primary language; only seeing it for a patient imported with an unknown primary language.

Reply:
Fixed!